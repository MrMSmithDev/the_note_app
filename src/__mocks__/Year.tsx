import { YearProps } from '@components/Year/Year';

export default function Year({ year }: YearProps) {
  return <div data-testid="mock-year">mock year</div>;
}
