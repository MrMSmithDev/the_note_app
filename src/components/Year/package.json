{
    "name": "year",
    "version": "1.0.0",
    "main": "Year.tsx",
    "dependencies": {
      "react": "^19.0.0"
    }
}