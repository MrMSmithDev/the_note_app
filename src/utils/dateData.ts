import Month from 'src/types/MonthType';

const dayData: string[] = ['S', 'M', 'T', 'W', 'T', 'F', 'S'];

const monthsData: Month[] = [
  { abbr: 'JAN', num: 1 },
  { abbr: 'FEB', num: 2 },
  { abbr: 'MAR', num: 3 },
  { abbr: 'APR', num: 4 },
  { abbr: 'MAY', num: 5 },
  { abbr: 'JUN', num: 6 },
  { abbr: 'JUL', num: 7 },
  { abbr: 'AUG', num: 8 },
  { abbr: 'SEP', num: 9 },
  { abbr: 'OCT', num: 10 },
  { abbr: 'NOV', num: 11 },
  { abbr: 'DEC', num: 12 },
];

export { dayData, monthsData };
