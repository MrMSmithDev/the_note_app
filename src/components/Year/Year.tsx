import React from 'react';
import { monthsData } from '@utils/dateData';
import MonthType from 'src/types/MonthType';
import Month from '@components/Month';

interface YearProps {
  year: number;
}

const Year: React.FC<YearProps> = ({ year }) => {
  return (
    <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-x-10 gap-y-3 h-1/2 mt-10">
      {monthsData.map((month: MonthType) => (
        <Month key={month.num} month={month} year={year} />
      ))}
    </div>
  );
};

export default Year;
