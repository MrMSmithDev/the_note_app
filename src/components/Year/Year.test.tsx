import { render, screen } from '@testing-library/react';
import Year from './Year';
import { monthsData } from '@utils/dateData';

jest.mock('@components/Month', () => require('@__mocks__/Month'));

describe('Year component', () => {
  const mockYear = 2025;

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders correctly to match snapshot', () => {
    const { asFragment } = render(<Year year={mockYear} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the month labels correctly', () => {
    render(<Year year={mockYear} />);

    monthsData.forEach((month) => {
      expect(screen.getByText(month.abbr)).toBeInTheDocument();
    });
  });

  it('renders the correct amount of months', () => {
    render(<Year year={mockYear} />);

    expect(screen.getAllByTestId('mock-month').length).toBe(12);
  });
});
