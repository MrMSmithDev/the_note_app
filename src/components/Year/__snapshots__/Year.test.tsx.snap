// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Year component renders correctly to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-x-10 gap-y-3 h-1/2 mt-10"
  >
    <div
      data-testid="mock-month"
    >
      JAN
    </div>
    <div
      data-testid="mock-month"
    >
      FEB
    </div>
    <div
      data-testid="mock-month"
    >
      MAR
    </div>
    <div
      data-testid="mock-month"
    >
      APR
    </div>
    <div
      data-testid="mock-month"
    >
      MAY
    </div>
    <div
      data-testid="mock-month"
    >
      JUN
    </div>
    <div
      data-testid="mock-month"
    >
      JUL
    </div>
    <div
      data-testid="mock-month"
    >
      AUG
    </div>
    <div
      data-testid="mock-month"
    >
      SEP
    </div>
    <div
      data-testid="mock-month"
    >
      OCT
    </div>
    <div
      data-testid="mock-month"
    >
      NOV
    </div>
    <div
      data-testid="mock-month"
    >
      DEC
    </div>
  </div>
</DocumentFragment>
`;
